TransferMethod/Sources/SelectTransferMethodTypePresenter.swift

```diff
@@ -310,14 +310,33 @@ final class SelectTransferMethodTypePresenter {
     }
 
     private func loadCurrency(_ keys: HyperwalletTransferMethodConfigurationKey?) {
-        guard let firstCurrency = keys?.currencies(from: selectedCountry)?.first,
-            let currencyCode = firstCurrency.code else {
-            view?.showAlert(message: String(format: "no_currency_available_error_message".localized(), selectedCountry))
-            return
+        guard let countries = keys?.countries(),
+              let country = countries.first(where: { $0.code == selectedCountry }) else {
+               // Handle the case when country is not found
+               let errorMessage = String(format: "no_country_available_error_message".localized(), selectedCountry ?? "Unknown Country")
+               view?.showAlert(message: errorMessage)
+               return
+           }
+        // Check if the country has a defaultCurrencyCode
+            if let defaultCurrencyCode = country.defaultCurrencyCode {
+                // If the defaultCurrencyCode exists, use it
+                selectedCurrency = defaultCurrencyCode
+            } else {
+                // If the defaultCurrencyCode is nil, fall back to the first currency in the currencyList
+                if let firstCurrency = keys?.currencies(from: selectedCountry)?.first,
+                   let currencyCode = firstCurrency.code {
+                    selectedCurrency = currencyCode
+                } else {
+                    // Show an alert if no currency is found
+                    let errorMessage = String(format: "no_currency_available_error_message".localized(), selectedCountry ?? "Unknown Country")
```

Comment:
The `Unknown Country` can't be used because the App is localized.  
- Please just use `view?.showAlert(message: String(format: "no_currency_available_error_message".localized(), selectedCountry))`